src/chains/ethereum/options/src/wallet-options.ts

```diff
@@ -181,45 +190,106 @@ export type WalletConfig = {
 export const WalletOptions: Definitions<WalletConfig> = {
   totalAccounts: {
     normalize,
+    shortDescription: "Number of accounts to generate at startup.",
     default: () => 10,
-    legacyName: "total_accounts"
+    legacyName: "total_accounts",
+    cliAliases: ["a", "accounts"],
+    cliType: "number"
   },
   accounts: {
-    normalize,
-    legacyName: "accounts"
+    normalize: rawInput => {
+      if (rawInput.length > 0) {
+        if (typeof rawInput[0] === "string") {
+          return (rawInput as string[]).map(accountString => {
+            const accountParts = accountString.split(",");
+            return {
+              secretKey: accountParts[0],
+              balance: accountParts[1]
+            } as OptionsAccount;
+          });
+        } else {
+          return rawInput as OptionsAccount[];
+        }
+      } else {
+        return [];
+      }
+    },
+    shortDescription:
+      "Array of Accounts. Each object should have a balance key with a hexadecimal value.",
+    legacyName: "accounts",
+    cliAliases: ["account"],
+    cliType: "array"
   },
   seed: {
     normalize,
-    default: () => randomAlphaNumericString(10, alea()),
-    legacyName: "seed"
+    shortDescription: "Seed to use to generate a mnemonic.",
+    default: config =>
```

Comment:
I just realized that my `default(config)` implementation depends on the order the options are specified in _this_ Definitions object.

The users options will be parsed in the order they appear in `Object.keys(WalletOptions)`.

Which would result in `seed` being parsed before `deterministic`, ensuring deterministic is _never_ set to `true`.

I haven't tested, but if my hunch here is true, the easiest thing to do is to just move `deterministic` right before `seed` in this `WalletOptions` object. And slap a big ol' comment in here about how the ordering is required (and test ensuring the correct wallet options get applied would be a nice bonus :-) ).

There are other more "technically correct" ways of solving this, but all the ones I've thought of have their own downsides and limitations (like maintaining dependency trees or a double pass over the options).

Reply:
Looking at the code in `packages/options/src/create.ts`, I concur with your hunch 👍 Thanks for the heads up!

Reply:
I also manually tested and confirmed that the order does matter (your hunch is correct)